aten/src/ATen/native/Assert.cpp

```diff
@@ -0,0 +1,23 @@
+#include <cstddef>
+#include <string>
+
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+void Assert(bool condition, std::string message) {
+  if (!condition) {
+    throw c10::AssertionError(message.c_str(), "");
+  }
+}
+
+void Assert(const Tensor& condition, std::string message) {
+  if (!at::native::all(condition).item<bool>()) {
```

Comment:
Does this match the boolean coercion rules for tensor? I don't think so.

```
>>> import torch
>>> bool(torch.ones(2))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Boolean value of Tensor with more than one value is ambiguous
```

Reply:
thanks, fixed